bridges/otelslog/exampe_test.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelslog_test
+
+import (
+	"log/slog"
+
+	"go.opentelemetry.io/contrib/bridges/otelslog"
+	"go.opentelemetry.io/otel/log/noop"
+)
+
+func Example() {
+	// Use a working LoggerProvider implementation instead e.g. using go.opentelemetry.io/otel/sdk/log.
+	provider := noop.NewLoggerProvider()
+
+	// Create an *slog.Logger with *otelslog.Handler and use it in your application.
+	slog.New(otelslog.NewHandler(otelslog.WithLoggerProvider(provider)))
```

Comment:
Should we use otelslog.NewLogger instead to simplify it?
```suggestion
	otelslog.NewLogger(otelslog.WithLoggerProvider(provider))
```

Reply:
I propose `otelslog.NewHandler` as most (all?) packages I am aware have only `Handler` factories. The reason is that one may want to use it together with some handler decorator e.g. https://github.com/samber/slog-multi. The `slog` package has `NewJSONHandler` and `NewTextHandler`.  See also: https://go.dev/wiki/Resources-for-slog

I even propose removing `otelslog.NewLogger` for sake of consistency and in order to remove redundant functionality.

@MrAlias WDYT?

Reply:
See https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5292 for the previous discussion.  I agree with the need for NewHandler for decorators.  I like the simplicity of being able to do `slog.SetDefault(otelslog.NewLogger())`, but I don't think NewLogger is strictly necessary.

Reply:
I think @dashpole is correct, the simplicity of `slog.SetDefault(otelslog.NewLogger())` is likely what most users are going to want.

Reply:
Personally, I lean towards smaller API surface. But it is just a preference.

Reply:
Applying suggestion and resolving